### ISSUE TITLE
[5.1] Plugin settings overwritten when hiding fields in the Field Manager

### ISSUE THREAD

**Issue by jeroenlammerts** [solspace/craft3-freeform#1231]
**Describe the bug or issue you're experiencing**

**Steps to reproduce**

1. Set a custom error message in /admin/freeform/settings/form-builder + save settings
2. Create a new form
3. Edit the field types (Field Manager)
4. Disable some field (groups)
5. Drag a field from Unassigned to Hidden + Save
6. Check modified project.yml

There is a POST call to /admin/freeform/api/fields/types/groups changing this values.
Some freeform settings are reverted back to defaults. For example errorMessage.

**Expected behavior**
Only hiddenFieldTypes should be added/modified in the project config configuration.

**Craft & Plugin Info (please complete the following information):**

- Craft Version: Craft Pro 4.8.0
- Freeform Version: 5.1.14
- Freeform Edition: Pro
- Fresh Install or Upgrade: Fresh install

**Additional context**
```
Application Info
PHP version	8.2.7
OS version	Linux 5.15.0-105-generic
Database driver & version	MySQL 8.0.36
Image driver & version	Imagick 3.7.0 (ImageMagick 6.9.10-23)
Craft edition & version	Craft Pro 4.8.0
Yii version	2.0.48.1
Twig version	v3.8.0
Guzzle version	7.8.1

Plugins
Cloudmailer	3.1.2
Elasticsearch	dev-master
Feed Me	dev-bugfix-1403
Formulieren	5.1.14
Incognito Field	1.3.0
Maps	4.0.4
Navigation	2.0.25
OpenStack Object Storage	3.0.5
Redactor	3.0.4
Redirect Manager	dev-master
SEO	4.0.42
Simple CSRF Refresh	2.0.0
Super Table	3.0.12
Typed link field	2.1.5
Typogrify	4.0.1
Varnish kube-httpcache	2.3.0
```

**Comment by johnvanhulsen** [comment]
The issue is larger than written above... 

When changing one field in `/freeform/settings/form-builder` and save this, go to another submenu item (eg. `/freeform/settings/form-behavior`) and save it (with or without changing any fields), the previous change in the form-builder is resetted, including all other earlier changed fields. 

Test this in 5.1.11 and 5.1.14 with Craft 4.8.9

**Comment by jeroenlammerts** [comment]
Also on production environments you can still move the fields to the hidden section in the field manager. When trying to save, it gives a 500 error.
Because allowAdminChanges is false in this env it cannot write the project config.

**Comment by kjmartens** [comment]
I'm very sorry for the trouble caused here, @jeroenlammerts and @johnvanhulsen.

We are investigating this now.

**Comment by kjmartens** [comment]
This is now corrected in Freeform 5.1.16+. 🙂
